data/tutorials/language/0lg_08_env.md

```diff
@@ -0,0 +1,557 @@
+```ocaml
+# let d = 21;;
+val d : int = 21
+
+# let d = 7 in d * 2;;
+val d : int = 14
+
+# d;;
+- : int = 21
+```
+
+## Function Values
+
+### Restrictions 
+
+Functions are supposed to be exactly as other values. However, there are two restrictions.
+
+Function values cannot be displayed in interactive sessions, but the placeholder `<fun>` is displayed. This is because there is nothing meaningful to print. Once parsed and typed-checked, OCaml discards the function's source code and nothing remains to be printed.
+```ocaml
+# sqrt;;
+- : float -> float = <fun>
+```
+
+Equality between functions can't be tested.
+```ocaml
+# pred = succ;;
+Exception: Invalid_argument "compare: functional value".
+```
+
+There are two main reasons explaining this:
+1. It is impossible to write an algorithm that takes two functions and returns `true` if they always return the same output when provided the same input and `false` otherwise
+1. Assuming it was possible, such an algorithm would declare that quicksort and bubble sort are equal. That would mean those procedures are substitutable, which is not the case.
+
+It may seem counterintuitive that classes of objects of the same kind (i.e., having the same type) exist where equality between objects does not make sense. High-school mathematics does not provide examples of those classes. But in the case of computing procedures seen as functions, equality isn't the right tool to compare them.
+
+### Calling Functions
+
+In OCaml, no symbol is needed to express passing an argument to a function. When several expressions are written side by side, the leftmost one should be a function. All the others are parameters. In OCaml, parentheses serve a single purpose: associating expressions to create subexpressions.
+```ocaml
+# max (21 * 2) (int_of_string "713");;
+- : int = 713
+```
+
+There are two alternative ways to call functions: using the application `@@` operator or the pipe `|>` operator.
+```ocaml
+# ( @@ );;
+- : ('a -> 'b) -> 'a -> 'b = <fun>
+
+# sqrt 9.0;;
+- : float = 3.
+
+# sqrt @@ 9.0;;
+```
+
+The application operator applies a parameter (on the right) to a function (on the left). It is useful when chaining several calls, as it avoids writing parentheses. Here is what it can look like:
+```ocaml
+# truncate (sqrt (float_of_int (int_of_string "81")));;
+- : int = 9
+
+# truncate @@ sqrt @@ float_of_int @@ int_of_string "81";;
+- : int = 9
+```
+
+The pipe operator also allows avoiding parentheses but in reversed order:
+```ocaml
+# ( |> );;
+- : 'a -> ('a -> 'b) -> 'b = <fun>
+
+# "81" |> int_of_string |> float_of_int |> sqrt |> truncate;;
+- : int = 9
+```
+This is just like a Unix shell pipe.
+
+### Anonymous Functions
+
+As citizens of the same level as other values, functions don't have to be bound to a name to exist (although some must, this will be explained later). Function values not bound to names are called _anonymous functions_. Here are a couple of examples:
```

Comment:
```suggestion
As citizens of the same level as other values, functions don't have to be bound to a name to exist (although some must, but this will be explained later). Function values not bound to names are called _anonymous functions_. Here are a couple of examples:
```